### PR TITLE
Meta-data: use the metadata package from egoscale v3

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
# Description

this adds the context to the HTTP request

## Checklist
(For exoscale contributors)

* [ ] Changelog updated (under *Unreleased* block)
* [ ] Testing

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,10 +6,11 @@ toolchain go1.22.5
 
 require (
 	github.com/exoscale/egoscale v0.102.3
+	github.com/exoscale/egoscale/v3 v3.1.4
```
Comment (sauterp) [inline go.mod:9]: why not use the laatest v3.1.7

File: exoscale/zones.go [hunk 2]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"fmt"
 
+	exometa "github.com/exoscale/egoscale/v3/metadata"
```
Comment (sauterp) [inline exoscale/zones.go:7]: you probably need to `go mod tidy && go mod vendor`

File: go.mod (hunk 1 above)
Comment (pierre-emmanuelJ) [reply go.mod:9]: This PR is pretty old 😄

**Comment by pierre-emmanuelJ** [comment]
fixed in https://github.com/exoscale/exoscale-cloud-controller-manager/pull/97

**Pull request closed without merging**